RFCs/0004-pluggable-monitor.md

```diff
@@ -0,0 +1,453 @@
++             "archiveFileName": "ble-mon-1.0.0-linux64.tar.gz",
++             "checksum": +SHA-256:0123456789abcdef0123456789abcdef0123456789abcdef",
++             "size": "12345678"
++           },
++           ...
++         ]
++       }
+      ],
+    }
+  }
+}
+```
+
+In this case we are adding an hypotetical `ble-monitor` version `1.0.0` to the toolset of the vendor `arduino`. From now on, we can uniquely refer to this monitor with the pair `PACKAGER` and `MONITOR_NAME`, in this case `arduino` and `ble-monitor` respectively.
+
+The compressed archive of the monitor must contain only a single executable file (the monitor itself) inside a single root folder. This is mandatory since the CLI will run this file automatically when a monitor instance is requested.
+
+#### Monitor tools integration
+
+Each core platform must refer to the specific monitor tools they need by adding them (together with the pluggable discoveries...) inside the `discoveryDependencies` field of the `package_index.json`:
+
+```diff
+{
+  "packages": [
+    {
+      "name": "arduino",
+      "maintainer": "Arduino",
+      "websiteURL": "http://www.arduino.cc/",
+
+      "platforms": [
+        {
+          "name": "Arduino AVR Boards",
+          "architecture": "avr",
+          "version": "1.6.2",
+          ...
+          "toolsDependencies": [
+            {
+              "packager": "arduino",
+              "name": "arm-none-eabi-gcc",
+              "version": "4.8.3-2014q1"
+            },
+            {
+              "packager": "arduino",
+              "name": "CMSIS",
+              "version": "4.5.0"
+            },
+            ...
+          ],
++         "discoveryDependencies": [  <--- Discoveries AND monitors used in the platform
+            {
+              "packager": "arduino",
+              "name": "ble-discovery"
+            },
++           {
++             "packager": "arduino",
++             "name": "ble-monitor"
++                                     <--- Version is not required!
++           }
+          ]
+        },
+
+        {
+          "name": "Arduino SAMD Boards",
+          "architecture": "samd",
+          "version": "1.6.18",
+          ...
+          "toolsDependencies": [ ... ],
+          "discoveryDependencies": [ ... ]
+        }
+
+      ],
+      "tools": [
+        {
+          "name": "arm-none-eabi-gcc",
+          "version": "4.8.3-2014q1",
+          "systems": [ ... ]
+        },
+        {
+          "name": "ble-discovery",
+          "version": "1.0.0",
+          "systems": [ ... ]
+        },
+        {
+          "name": "ble-monitor",
+          "version": "1.0.0",
+          "systems": [ ... ]
+        }
+      ],
+    }
+  }
+}
+```
+
+Adding the needed monitor tools in the `discoveryDependencies` allows the CLI to install them together with the platform. Also, differently from the other `toolsDependencies`, the version is not required since it will always be used the latest version available.
```

Comment:
```suggestion
Adding the needed monitor tools in the `discoveryDependencies` allows the CLI to install them together with the platform. Also, differently from the other `toolsDependencies`, the version is not required since the latest version available will always be used.
```